sdk/storage/Azure.Storage.Common/src/TaskExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace Azure.Storage
+{
+    /// <summary>
+    /// Extensions to execute async Tasks synchronously.
+    /// </summary>
+    internal static class TaskExtensions
+    {
+        /// <summary>
+        /// Ensure the Task has finished executing.
+        /// </summary>
+        /// <typeparam name="T">Task's return type.</typeparam>
+        /// <param name="task">The task.</param>
+        /// <returns>The result of executing the task.</returns>
+        public static T EnsureCompleted<T>(this Task<T> task)
```

Comment:
We need significant testing here, please, in a variety of environments (ASP.NET, desktop, WPF, WinForms, WebForms, anything that might play with SynchronizationContext).

Reply:
Filed #6759 to make sure we do this.